Geometry/CaloGeometry/interface/CaloCellGeometryHGCALAdapter.h

```diff
@@ -0,0 +1,16 @@
+#ifndef GeometryCaloCellGeometryHGCALAdapter
+#define GeometryCaloCellGeometryHGCALAdapter
+
+#include "Geometry/CaloGeometry/interface/FlatTrd.h"
+
+class CaloCellGeometryHGCALAdapter : public FlatTrd {
+public:
+  explicit CaloCellGeometryHGCALAdapter (const FlatTrd * f, GlobalPoint p) : FlatTrd(*f), position_(p) {}
```

Comment:
why is a copy of the FlatTrd object needed here (rather than keeping the pointer) - sorry to add noise.

Reply:
@davidlange6 the `PFRecHit` has a member `CaloCellGeometry const * caloCell_`. We have to provide it with something equivalent - this new class inherits from `FlatTrd` which inherits from `CaloCellGeometry`, but returns the correct position. I thought about it for a while and couldn't come up with a better hack (i.e. without changing the `PFRecHit` class).

Reply:
Ok so why does this not work? [I think the savings is not so substantial as CaloCellGeometry is the majority of the size of FlatTrd so lets save the question for later 

class CaloCellGeometryHGCalAdapter : public CaloCellGeometry { 

public:
  ...

private:
  const FlatTrd *trd_;
  GlobalPoint position_;